internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go

```diff
@@ -0,0 +1,241 @@
+package cluster_mgrs
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	fleeterrors "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/golang/glog"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/workers"
+	"github.com/google/uuid"
+)
+
+const (
+	dynamicScaleDownWorkerType = "dynamic_scale_down"
+)
+
+type DynamicScaleDownManager struct {
+	workers.BaseWorker
+
+	dataplaneClusterConfig *config.DataplaneClusterConfig
+	clusterProvidersConfig *config.ProviderConfig
+	kafkaConfig            *config.KafkaConfig
+	clusterService         services.ClusterService
+}
+
+var _ workers.Worker = &DynamicScaleDownManager{}
+
+func NewDynamicScaleDownManager(
```

Comment:
Test will be added later, for now we want to nail the scale down detection logic

Reply:
I've added unit test and updated the integration test